pyproject.toml

```diff
@@ -52,7 +54,10 @@ lockfile = "^0.12.2"
 pylint = "*"
 pytest = "*"
 pytest-cov = "*"
-scipy = "~1.10.1"
+scipy = [
+    {version="~1.10.1", python= "<3.12"},
+    {version=">=1.12.0", python= ">=3.12"}
```

Comment:
if possible, just always use SciPy 1.12, not matter what the version of Python
(according to [SciPy 1.12.0 release notes](https://docs.scipy.org/doc/scipy/release/1.12.0-notes.html#scipy-1-12-0-release-notes), SciPy 1.12 supports Python 3.9+)

Reply:
ok, we can take care of it later